packages/traceability/issuer-api/migrations/1633507266818-StoreTxHashInsteadBlock.ts

```diff
@@ -0,0 +1,105 @@
+import { Certificate, Contracts, IBlockchainProperties } from '@energyweb/issuer';
+import { getProviderWithFallback } from '@energyweb/utils-general';
+import { Wallet } from 'ethers';
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+import { BlockchainProperties } from '../src/pods/blockchain/blockchain-properties.entity';
+
+export class StoreTxHashInsteadBlock1633507266818 implements MigrationInterface {
+    name = 'StoreTxHashInsteadBlock1633507266818';
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(
+            `ALTER TABLE "issuer_certificate" ADD "creationTransactionHash" character varying`
+        );
+        const existingCertificates = await queryRunner.query(`SELECT id FROM "issuer_certificate"`);
+        console.log({
+            existingCertificates
+        });
+
+        if (existingCertificates.length > 0) {
+            const blockchainProperties: BlockchainProperties = await queryRunner.query(
+                `SELECT * FROM "issuer_blockchain_properties"`
+            );
+            const wrapped: IBlockchainProperties =
+                this.getBlockchainProperties(blockchainProperties);
+
+            const syncedCertificates = await Promise.all(
+                existingCertificates.map((cert: any) => new Certificate(cert.id, wrapped).sync)
```

Comment:
How exausting this may be? For one case we can have around 30k certificates before we even execute this.

Reply:
Depends on the RPC node entirely, but you're right, it might trigger some DDoS protection.